server/lib/utils/lexicographic_ordinal.ts

```diff
@@ -0,0 +1,68 @@
+import { maxBy } from 'lodash-es'
+import { characters, findOrdinalBetween } from '#lib/find_ordinal'
+import type { ListingElement } from '#types/element'
+
+export function findNewOrdinal (element, list, inclusiveOrdinal) {
+  // Format from a more human readable "inclusive" order (starting by 1, 2, 3)
+  // to more bot readable "exclusive" order (starting by 0, 1, 2)
+  // see http://kilby.stanford.edu/~rvg/ordinal.html
+  let newOrdinal = inclusiveOrdinal - 1
+
+  // Place element in last position if newOrdinal is too high
+  if (list.length < newOrdinal) newOrdinal = list.length - 1
+  if (list[newOrdinal]._id === element._id) return
+  removeElementIfNecessary(element, list, newOrdinal)
+  return findNewLexicographicOrdinal(newOrdinal, list)
+}
+
+function removeElementIfNecessary (element, list, newOrdinal) {
+  // A make-it-fast implementation: find index from a subset of element,
+  // as element must be removed only if its being moved up in the list
+  // hence only if element is between first element and newOrdinal element
+  const listSubset = list.slice(0, newOrdinal)
+  const index = listSubset.findIndex(el => el._id === element._id)
+  if (index !== -1) list.splice(index, 1)
+}
+
+function findNewLexicographicOrdinal (newOrdinal, currentElements) {
```

Comment:
[non-blocking] As this juggles between numeric and lexicographic ordinals, adding types would help to make it more readable.